sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -1213,8 +1213,8 @@ private async Task RunProcessingAsync(CancellationToken cancellationToken)
                 {
                     if (!ActivePartitionProcessors.TryGetValue(partitionId, out var partitionProcessor) || partitionProcessor.ProcessingTask.IsCompleted)
                     {
-                        await StopProcessingPartitionAsync(partitionId, ProcessingStoppedReason.OwnershipLost, cancellationToken).ConfigureAwait(false);
-                        await StartProcessingPartitionAsync(partitionId, cancellationToken).ConfigureAwait(false);
+                        await TryStopProcessingPartitionAsync(partitionId, ProcessingStoppedReason.OwnershipLost, cancellationToken).ConfigureAwait(false);
+                        await TryStartProcessingPartitionAsync(partitionId, cancellationToken).ConfigureAwait(false);
```

Comment:
@christothes :  This is the bit that I brought up with in our conversation today.   I'm thinking that it makes sense to abandon the partition at this point, since the processor will have applied a robust set of retries, including use of a fresh consumer.   If we hit this block, the processor gave up and believes that we're not making forward progress.   Currently, we'll just restart at the beginning; if we short-circuit here and relinquish the partition, we would potentially give other processors the chance to take ownership and allow it to move forward.

That said, I could be overlooking some logic at the load balancing level that does this already.   I'd appreciate your thoughts, if you'd be so kind.

Reply:
I don't think you are missing anything - we don't currently handle commands about ownership at all in the load balancer as I think the original intention was to encapsulate the concern of ownership decisions fully. That said, the load balancer has no insight to processing faults so it wouldn't have any context to make an abandon decision on its own.

My only concern about abandoning a "poison partition", regardless of which component owns the decision, is that under a stabilized load balance state the only processor that may attempt to re-claim it would be same one that abandoned it. I'm leaning towards the status quo of offloading the "poison partition" failure to the developer, in that we're most likely still doing that even if we actively abandon. 

Thoughts?

Reply:
I agree; between conversation here and in our offline chat, I think it makes sense to punt this for now so that we can think things through and come up with a more comprehensive approach should this be a problem that is observed in real-world scenarios.

As you mention, between the initialize and stop handlers, developers using the processor do have enough information to detect a poison pattern and react in the way that is appropriate for their scenario;  I'll open an issue on the backlog for future consideration.